state-synchronizer/src/coordinator.rs

```diff
@@ -315,13 +328,55 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
     /// The function is called after new txns have been applied to the local storage.
     /// As a result it might:
     /// 1) help remote subscribers with long poll requests, 2) finish local sync request
-    async fn process_commit(&mut self) -> Result<()> {
+    async fn process_commit(
+        &mut self,
+        transactions: Vec<CommittedTransaction>,
+        callback_consensus: Option<oneshot::Sender<Result<CommitResponse>>>,
+    ) -> Result<()> {
         // We choose to re-sync the state with the storage as it's the simplest approach:
         // in case the performance implications of re-syncing upon every commit are high,
         // it's possible to manage some of the highest known versions in memory.
         self.sync_state_with_local_storage().await?;
         let local_version = self.local_state.highest_version_in_local_storage();
         counters::COMMITTED_VERSION.set(local_version as i64);
+        let block_timestamp_usecs = self
+            .local_state
+            .highest_local_li
+            .ledger_info()
+            .timestamp_usecs();
+
+        // send notif to shared mempool
+        let (callback, callback_rcv) = oneshot::channel();
+        let req = CommitNotification {
+            transactions,
+            block_timestamp_usecs,
+            callback,
+        };
+        let mut mempool_channel = self.mempool_channel.clone();
+        tokio::spawn(async move {
+            if let Err(e) = mempool_channel.try_send(req) {
+                error!(
+                    "[state sync] failed to send commit notif to shared mempool: {:?}",
+                    e
+                );
+            }
+            if let Err(e) = timeout_at(Instant::now() + Duration::from_secs(1), callback_rcv).await
+            {
+                error!(
+                        "[state sync] did not receive ACK for commit notification sent to mempool: {:?}", e
+                        )
+            }
+
+            if let Some(cb) = callback_consensus {
```

Comment:
nit: may be we should rename the variable from `callback_consensus` to some more generic `commit_callback`. You're technically right that the commit request is coming from consensus, but it's still confusing because there is a `sync_request_callback` below, which is for consensus as well :).